File: examples/python/filmes_series.py
"""
Cognima API - Exemplo de Filmes e Séries (XC IPTV)

Este exemplo demonstra como usar a API de filmes e séries
integrada com XC IPTV.
"""

import requests
from typing import Optional, List, Dict

# Configuração
API_BASE_URL = 'https://cog.api.br/api/v1'
API_KEY = 'sua-api-key-aqui'  # Substitua pela sua API key

# Headers
HEADERS = {
    'X-API-Key': API_KEY,
    'Content-Type': 'application/json'
}

# ===== EXEMPLOS DE FILMES =====

def listar_categorias_filmes() -> Optional[List[Dict]]:
    """Exemplo 1: Listar todas as categorias de filmes"""
    try:
        print('\n📁 Listando categorias de filmes...\n')
        
        response = requests.get(
            f'{API_BASE_URL}/filmes/categorias',
            headers=HEADERS
        )
        response.raise_for_status()
        
        categorias = response.json()['data']
        
        print(f'✅ {len(categorias)} categorias encontradas:')
        for cat in categorias[:5]:
            print(f"   - {cat['category_name']} (ID: {cat['category_id']})")
        
        return categorias
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')
        return None


def listar_filmes(category_id: Optional[str] = None) -> Optional[List[Dict]]:
    """Exemplo 2: Listar filmes (todos ou por categoria)"""
    try:
        print('\n🎬 Listando filmes...\n')
        
        params = {'category_id': category_id} if category_id else {}
        response = requests.get(
            f'{API_BASE_URL}/filmes',
            headers=HEADERS,
            params=params
        )
        response.raise_for_status()
        
        filmes = response.json()['data']
        
        print(f'✅ {len(filmes)} filmes encontrados:')
        for filme in filmes[:5]:
            print(f"   - {filme['name']}")
            print(f"     ID: {filme['stream_id']} | Rating: {filme.get('rating', 'N/A')}")
        
        return filmes
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')
        return None


def buscar_filmes(query: str) -> Optional[List[Dict]]:
    """Exemplo 3: Buscar filmes por nome"""
    try:
        print(f'\n🔍 Buscando filmes: "{query}"...\n')
        
        response = requests.get(
            f'{API_BASE_URL}/filmes/buscar',
            headers=HEADERS,
            params={'query': query}
        )
        response.raise_for_status()
        
        filmes = response.json()['data']
        
        print(f'✅ {len(filmes)} resultados encontrados:')
        for filme in filmes:
            print(f"   - {filme['name']}")
            print(f"     ID: {filme['stream_id']} | Rating: {filme.get('rating', 'N/A')}")
        
        return filmes
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')
        return None


def obter_detalhes_filme(stream_id: str) -> Optional[Dict]:
    """Exemplo 4: Obter detalhes completos de um filme"""
    try:
        print(f'\n📄 Obtendo detalhes do filme ID {stream_id}...\n')
        
        response = requests.get(
            f'{API_BASE_URL}/filmes/{stream_id}',
            headers=HEADERS
        )
        response.raise_for_status()
        
        dados = response.json()['data']
        info = dados['info']
        
        print('✅ Detalhes do filme:')
        print(f"   Nome: {info.get('name', 'N/A')}")
        print(f"   Gênero: {info.get('genre', 'N/A')}")
        print(f"   Rating: {info.get('rating', 'N/A')}")
        print(f"   Duração: {info.get('duration', 'N/A')}")
        print(f"   Diretor: {info.get('director', 'N/A')}")
        print(f"   Elenco: {info.get('cast', 'N/A')}")
        print(f"   Sinopse: {info.get('plot', 'N/A')}")
        print(f"   URL de Stream: {dados.get('streamUrl', 'N/A')}")
        
        return dados
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')
        return None


# ===== EXEMPLOS DE SÉRIES =====

def listar_categorias_series() -> Optional[List[Dict]]:
    """Exemplo 5: Listar categorias de séries"""
    try:
        print('\n📁 Listando categorias de séries...\n')
        
        response = requests.get(
            f'{API_BASE_URL}/series/categorias',
            headers=HEADERS
        )
        response.raise_for_status()
        
        categorias = response.json()['data']
        
        print(f'✅ {len(categorias)} categorias encontradas:')
        for cat in categorias[:5]:
            print(f"   - {cat['category_name']} (ID: {cat['category_id']})")
        
        return categorias
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')
        return None


def buscar_series(query: str) -> Optional[List[Dict]]:
    """Exemplo 6: Buscar séries"""
    try:
        print(f'\n🔍 Buscando séries: "{query}"...\n')
        
        response = requests.get(
            f'{API_BASE_URL}/series/buscar',
            headers=HEADERS,
            params={'query': query}
        )
        response.raise_for_status()
        
        series = response.json()['data']
        
        print(f'✅ {len(series)} resultados encontrados:')
        for serie in series[:5]:
            print(f"   - {serie['name']}")
            print(f"     ID: {serie['series_id']} | Rating: {serie.get('rating', 'N/A')}")
        
        return series
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')
        return None


def obter_detalhes_serie(series_id: str) -> Optional[Dict]:
    """Exemplo 7: Obter detalhes completos de uma série (com temporadas e episódios)"""
    try:
        print(f'\n📄 Obtendo detalhes da série ID {series_id}...\n')
        
        response = requests.get(
            f'{API_BASE_URL}/series/{series_id}',
            headers=HEADERS
        )
        response.raise_for_status()
        
        dados = response.json()['data']
        info = dados['info']
        
        print('✅ Detalhes da série:')
        print(f"   Nome: {info.get('name', 'N/A')}")
        print(f"   Gênero: {info.get('genre', 'N/A')}")
        print(f"   Rating: {info.get('rating', 'N/A')}")
        print(f"   Temporadas: {len(dados.get('seasons', []))}")
        
        if 'episodes' in dados and dados['episodes']:
            total_episodes = sum(len(eps) for eps in dados['episodes'].values())
            print(f"   Total de Episódios: {total_episodes}")
            
            # Mostrar primeira temporada
            first_season = sorted(dados['episodes'].keys())[0]
            print(f"\n   📺 Temporada {first_season}:")
            for ep in dados['episodes'][first_season][:3]:
                print(f"      E{ep['episode_num']}: {ep.get('title', 'Sem título')}")
                print(f"      Stream URL: {ep.get('streamUrl', 'N/A')}")
        
        return dados
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')
        return None


def gerar_link_player(tipo: str, id: str, episode_id: Optional[str] = None) -> str:
    """Exemplo 8: Gerar link do player web"""
    BASE_URL = 'https://cog.api.br'
    
    if tipo == 'filme':
        link = f'{BASE_URL}/watch/{id}'
        print(f'\n🎬 Link do Player (Filme):\n   {link}\n')
        return link
    elif tipo == 'serie':
        if episode_id:
            link = f'{BASE_URL}/watch/series/{id}/{episode_id}'
        else:
            link = f'{BASE_URL}/watch/series/{id}'
        print(f'\n📺 Link do Player (Série):\n   {link}\n')
        return link


# ===== EXECUTAR EXEMPLOS =====

def executar_exemplos():
    """Executar todos os exemplos"""
    print('═══════════════════════════════════════════════════')
    print('   🎬 Cognima API - Exemplos de Filmes e Séries')
    print('═══════════════════════════════════════════════════')

    # Exemplos de filmes
    listar_categorias_filmes()
    listar_filmes()
    
    filmes_matrix = buscar_filmes('matrix')
    if filmes_matrix and len(filmes_matrix) > 0:
        obter_detalhes_filme(filmes_matrix[0]['stream_id'])
        gerar_link_player('filme', filmes_matrix[0]['stream_id'])

    # Exemplos de séries
    listar_categorias_series()
    
    series_breaking = buscar_series('breaking bad')
    if series_breaking and len(series_breaking) > 0:
        detalhes = obter_detalhes_serie(series_breaking[0]['series_id'])
        gerar_link_player('serie', series_breaking[0]['series_id'])
        
        # Link para episódio específico
        if detalhes and 'episodes' in detalhes and detalhes['episodes']:
            first_season = sorted(detalhes['episodes'].keys())[0]
            if detalhes['episodes'][first_season]:
                episode_id = detalhes['episodes'][first_season][0]['id']
                gerar_link_player('serie', series_breaking[0]['series_id'], episode_id)

    print('\n═══════════════════════════════════════════════════')
    print('   ✅ Exemplos concluídos!')
    print('═══════════════════════════════════════════════════\n')


if __name__ == '__main__':
    executar_exemplos()

File: examples/nodejs/filmes-series.js
/**
 * Cognima API - Exemplo de Filmes e Séries (XC IPTV)
 * 
 * Este exemplo demonstra como usar a API de filmes e séries
 * integrada com XC IPTV.
 */

const axios = require('axios');

// Configuração
const API_BASE_URL = 'https://cog.api.br/api/v1';
const API_KEY = 'sua-api-key-aqui'; // Substitua pela sua API key

// Cliente HTTP configurado
const api = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    'X-API-Key': API_KEY,
    'Content-Type': 'application/json'
  }
});

// ===== EXEMPLOS DE FILMES =====

/**
 * Exemplo 1: Listar todas as categorias de filmes
 */
async function listarCategoriasFilmes() {
  try {
    console.log('\n📁 Listando categorias de filmes...\n');
    
    const response = await api.get('/filmes/categorias');
    const categorias = response.data.data;
    
    console.log(`✅ ${categorias.length} categorias encontradas:`);
    categorias.slice(0, 5).forEach(cat => {
      console.log(`   - ${cat.category_name} (ID: ${cat.category_id})`);
    });
    
    return categorias;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Exemplo 2: Listar filmes (todos ou por categoria)
 */
async function listarFilmes(categoryId = null) {
  try {
    console.log('\n🎬 Listando filmes...\n');
    
    const params = categoryId ? { category_id: categoryId } : {};
    const response = await api.get('/filmes', { params });
    const filmes = response.data.data;
    
    console.log(`✅ ${filmes.length} filmes encontrados:`);
    filmes.slice(0, 5).forEach(filme => {
      console.log(`   - ${filme.name}`);
      console.log(`     ID: ${filme.stream_id} | Rating: ${filme.rating || 'N/A'}`);
    });
    
    return filmes;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Exemplo 3: Buscar filmes por nome
 */
async function buscarFilmes(query) {
  try {
    console.log(`\n🔍 Buscando filmes: "${query}"...\n`);
    
    const response = await api.get('/filmes/buscar', {
      params: { query }
    });
    const filmes = response.data.data;
    
    console.log(`✅ ${filmes.length} resultados encontrados:`);
    filmes.forEach(filme => {
      console.log(`   - ${filme.name}`);
      console.log(`     ID: ${filme.stream_id} | Rating: ${filme.rating || 'N/A'}`);
    });
    
    return filmes;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Exemplo 4: Obter detalhes completos de um filme
 */
async function obterDetalhesFilme(streamId) {
  try {
    console.log(`\n📄 Obtendo detalhes do filme ID ${streamId}...\n`);
    
    const response = await api.get(`/filmes/${streamId}`);
    const dados = response.data.data;
    
    console.log('✅ Detalhes do filme:');
    console.log(`   Nome: ${dados.info.name}`);
    console.log(`   Gênero: ${dados.info.genre || 'N/A'}`);
    console.log(`   Rating: ${dados.info.rating || 'N/A'}`);
    console.log(`   Duração: ${dados.info.duration || 'N/A'}`);
    console.log(`   Diretor: ${dados.info.director || 'N/A'}`);
    console.log(`   Elenco: ${dados.info.cast || 'N/A'}`);
    console.log(`   Sinopse: ${dados.info.plot || 'N/A'}`);
    console.log(`   URL de Stream: ${dados.streamUrl}`);
    
    return dados;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

// ===== EXEMPLOS DE SÉRIES =====

/**
 * Exemplo 5: Listar categorias de séries
 */
async function listarCategoriasSeries() {
  try {
    console.log('\n📁 Listando categorias de séries...\n');
    
    const response = await api.get('/series/categorias');
    const categorias = response.data.data;
    
    console.log(`✅ ${categorias.length} categorias encontradas:`);
    categorias.slice(0, 5).forEach(cat => {
      console.log(`   - ${cat.category_name} (ID: ${cat.category_id})`);
    });
    
    return categorias;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Exemplo 6: Buscar séries
 */
async function buscarSeries(query) {
  try {
    console.log(`\n🔍 Buscando séries: "${query}"...\n`);
    
    const response = await api.get('/series/buscar', {
      params: { query }
    });
    const series = response.data.data;
    
    console.log(`✅ ${series.length} resultados encontrados:`);
    series.slice(0, 5).forEach(serie => {
      console.log(`   - ${serie.name}`);
      console.log(`     ID: ${serie.series_id} | Rating: ${serie.rating || 'N/A'}`);
    });
    
    return series;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Exemplo 7: Obter detalhes completos de uma série (com temporadas e episódios)
 */
async function obterDetalhesSerie(seriesId) {
  try {
    console.log(`\n📄 Obtendo detalhes da série ID ${seriesId}...\n`);
    
    const response = await api.get(`/series/${seriesId}`);
    const dados = response.data.data;
    
    console.log('✅ Detalhes da série:');
    console.log(`   Nome: ${dados.info.name}`);
    console.log(`   Gênero: ${dados.info.genre || 'N/A'}`);
    console.log(`   Rating: ${dados.info.rating || 'N/A'}`);
    console.log(`   Temporadas: ${dados.seasons?.length || 0}`);
    
    if (dados.episodes) {
      const totalEpisodes = Object.values(dados.episodes).reduce((sum, eps) => sum + eps.length, 0);
      console.log(`   Total de Episódios: ${totalEpisodes}`);
      
      // Mostrar primeira temporada
      const firstSeason = Object.keys(dados.episodes)[0];
      if (firstSeason) {
        console.log(`\n   📺 Temporada ${firstSeason}:`);
        dados.episodes[firstSeason].slice(0, 3).forEach(ep => {
          console.log(`      E${ep.episode_num}: ${ep.title || 'Sem título'}`);
          console.log(`      Stream URL: ${ep.streamUrl}`);
        });
      }
    }
    
    return dados;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Exemplo 8: Gerar link do player web
 */
function gerarLinkPlayer(tipo, id, episodeId = null) {
  const BASE_URL = 'https://cog.api.br';
  
  if (tipo === 'filme') {
    const link = `${BASE_URL}/watch/${id}`;
    console.log(`\n🎬 Link do Player (Filme):\n   ${link}\n`);
    return link;
  } else if (tipo === 'serie') {
    const link = episodeId 
      ? `${BASE_URL}/watch/series/${id}/${episodeId}`
      : `${BASE_URL}/watch/series/${id}`;
    console.log(`\n📺 Link do Player (Série):\n   ${link}\n`);
    return link;
  }
}

// ===== EXECUTAR EXEMPLOS =====

async function executarExemplos() {
  console.log('═══════════════════════════════════════════════════');
  console.log('   🎬 Cognima API - Exemplos de Filmes e Séries');
  console.log('═══════════════════════════════════════════════════');

  // Exemplos de filmes
  await listarCategoriasFilmes();
  await listarFilmes();
  
  const filmesMatrix = await buscarFilmes('matrix');
  if (filmesMatrix && filmesMatrix.length > 0) {
    await obterDetalhesFilme(filmesMatrix[0].stream_id);
    gerarLinkPlayer('filme', filmesMatrix[0].stream_id);
  }

  // Exemplos de séries
  await listarCategoriasSeries();
  
  const seriesBreaking = await buscarSeries('breaking bad');
  if (seriesBreaking && seriesBreaking.length > 0) {
    const detalhes = await obterDetalhesSerie(seriesBreaking[0].series_id);
    gerarLinkPlayer('serie', seriesBreaking[0].series_id);
    
    // Link para episódio específico
    if (detalhes && detalhes.episodes) {
      const firstSeason = Object.keys(detalhes.episodes)[0];
      if (firstSeason && detalhes.episodes[firstSeason][0]) {
        gerarLinkPlayer('serie', seriesBreaking[0].series_id, detalhes.episodes[firstSeason][0].id);
      }
    }
  }

  console.log('\n═══════════════════════════════════════════════════');
  console.log('   ✅ Exemplos concluídos!');
  console.log('═══════════════════════════════════════════════════\n');
}

// Executar se for o arquivo principal
if (require.main === module) {
  executarExemplos().catch(console.error);
}

// Exportar funções para uso em outros arquivos
module.exports = {
  listarCategoriasFilmes,
  listarFilmes,
  buscarFilmes,
  obterDetalhesFilme,
  listarCategoriasSeries,
  buscarSeries,
  obterDetalhesSerie,
  gerarLinkPlayer
};
